crates/build/src/workspace/manifest.rs

```diff
@@ -548,6 +602,67 @@ fn crate_type_exists(crate_type: &str, crate_types: &[value::Value]) -> bool {
         .any(|v| v.as_str().map_or(false, |s| s == crate_type))
 }
 
+fn merge_workspace_with_crate_dependencies(
+    section_name: &str,
+    crate_toml: &mut value::Table,
+    workspace_dependencies: &value::Table,
+) -> Result<()> {
+    let Some(dependencies) = crate_toml.get_mut(section_name) else {
+        return Ok(())
+    };
+
+    let table = dependencies
+        .as_table_mut()
+        .ok_or_else(|| anyhow::anyhow!("dependencies should be a table"))?;
+
+    for (name, value) in table {
+        let Some(dependency) = value.as_table_mut() else {
+            continue
+        };
+
+        let is_workspace_dependency = dependency
+            .get_mut("workspace")
+            .unwrap_or(&mut toml::Value::Boolean(false))
+            .as_bool()
+            .unwrap_or(false);
+        if !is_workspace_dependency {
+            continue
+        }
+
+        let workspace_dependency = workspace_dependencies.get(name).unwrap();
```

Comment:
Please replace the `unwrap` with an `ok_or_else` for consistency with other checks for invalid manifests.

Reply:
Done